### PR TITLE
New serverless pattern - apigw-http-api-lambda-ts-cdk

### PR DESCRIPTION
**Pull request by philippewanner**
# apigw-http-api-lambda-ts-cdk

## Description

This pattern create an Amazon API Gateway HTTP API and a dummy Lambda function returning a simple "Hello World".

## Language
Node.js (Typescript)

## Framework
CDK

## Service from/to
Service from: API Gateway HTTP API (v2)
Service to: Lambda

## Description
This pattern creates an Amazon API Gateway HTTP API (v2) that integrates with an AWS Lambda function by using AWS Cloud Development Kit (AWS CDK).

Amazon API Gateway HTTP API is designed with minimal features so it can be offered at a lower price than the REST API. For more information, please read the official document [Choosing between REST APIs and HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-vs-rest.html).

This pattern deploys a API Gateway HTTP API that is responsible to route the root path ("/") to the AWS Lambda function. The function (called 'dummy') return a 200 HTTP code and "Hello World" payload to the caller.

The template includes a CDK stack a HTTP API and a Lambda resources. The stack will output the HTTP API URL after the deployment, so you can test the API.

## Deployment commands
npm run deploy

## Payload example (e.g. Lambda event payload from source service).

N/A

## Additional resources (optional: link and anchor text, up to 5 resources)

    [Working with the AWS CDK in TypeScript](https://docs.aws.amazon.com/cdk/v2/guide/work-with-cdk-typescript.html)
    [Working with HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api.html)
    [Building Lambda functions with TypeScript](https://docs.aws.amazon.com/lambda/latest/dg/lambda-typescript.html)

## Author bio

Name: Philippe Wanner
Photo URL: https://www.linkedin.com/in/philippe-wanner/
LinkedIn: https://www.linkedin.com/in/philippe-wanner/
Twitter handle: n/a
Description (up to 255 chars): Philippe is a Senior Specialist Solutions Architect at Amazon Web Services based in Zurich, Switzerland. His role is to spread the migration and modernization best practices for large organisations. 
[Follow on LinkedIn](https://www.linkedin.com/in/philippe-wanner/)

### CODE REVIEW

File: apigw-http-api-lambda-ts-cdk/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Amazon API Gateway HTTP API with AWS Lambda integration
+
+This pattern in CDK offers a boilerlate to generate an Amazon API Gateway HTTP API endpoint with "ANY" method from the specified path. The Lambda function provided in TypeScript only returns a dummy "HelloWorld".
+
+If Amazon API Gateway HTTP API (v2) covers all the features' requirements, it's recommended to use HTTP API (v2) over REST API (v1). The HTTP API has the advantage to be significantly less expensive, and it peforms better.
```
Comment (ellisms) [inline apigw-http-api-lambda-ts-cdk/README.md:5]: Can you quantify the savings and performance? As written, the statement is vague.

File: apigw-http-api-lambda-ts-cdk/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Learn more about this pattern at Serverless Land Patterns: [https://serverlessland.com/patterns/apigw-http-api-lambda-ts-cdk](https://serverlessland.com/patterns/apigw-http-api-lambda-ts-cdk)
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
```
Comment (ellisms) [inline apigw-http-api-lambda-ts-cdk/README.md:11]: From my test, it looks like docker is also required. I would add it to the list.

File: apigw-http-api-lambda-ts-cdk/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+          "deploy": "cdk deploy --profile <YOUR_PROFILE_NAME>",
+          "destroy": "cdk destroy --profile <YOUR_PROFILE_NAME>"
+        }
+      }
+    ```
+
+## How it works
+
+The solution works as follow:
+1. CDK will provision an Amazon Gateway API HTTP API and a dummy Lambda function.
```
Comment (ellisms) [inline apigw-http-api-lambda-ts-cdk/README.md:50]:
```suggestion
1. CDK will provision an Amazon API Gateway HTTP API and a sample Lambda function.
```

File: apigw-http-api-lambda-ts-cdk/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+2. It will integrate the Lambda function to the  HTTP API with the route "/"
```
Comment (ellisms) [inline apigw-http-api-lambda-ts-cdk/README.md:51]:
```suggestion
2. It will integrate the Lambda function with the  HTTP API using the route "/"
```

**Review by ellisms — CHANGES_REQUESTED** [review]
Hi @philippewanner - thanks for the submission. Requesting a few changes before we can proceed.

**Comment by philippewanner** [comment]
The file example-pattnern.json has been added as requested

**Pull request merged**